src/java/org/apache/cassandra/index/sai/disk/PrimaryKeyMap.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+import org.apache.cassandra.index.sai.SSTableQueryContext;
+import org.apache.cassandra.index.sai.utils.PrimaryKey;
+
+/**
+ * A bidirectional map of {@link PrimaryKey} to row ID. Implementations of this interface
+ * are not expected to be threadsafe.
```

Comment:
So we can annotate with `@NotThreadSafe`

Reply:
The implementation in `InvertedIndexSearcherTest` _is_ thread-safe, right? I'd rather avoid annotating interfaces, especially if what they indicate is only partially true.